### DISCUSSION TITLE
How to get rid of the last piece of stringing left from the last printed layer?

### DISCUSSION THREAD

**Discussion by earthgrazer** [bassamanator/Sovol-SV06-firmware#130] [General]
The PRINT_END macro leaves substantial stringing that requires post-processing with a cutter to remove.

**To Reproduce**
1. Lay out two copies of any small model on the plate in slicer.
2. Set print settings to print each object on the plate individually. This is to demonstrate the stringing issue comes from the PRINT_END macro. In PrusaSlicer, this is the "Complete individual objects" setting.
3. Start and finish print.

**Expected behavior**
The object at the end of the print should have minimal stringing from where the nozzle leaves the last layer.

**Screenshots**
N/A

**What printer are you using?**

SV06

**What kind of device are you using to host Klipper?**

Raspberry Pi 2b.

**How did you install Klipper?**

MainsailOS.

**Is your local copy of this configuration up to date?**

2023-12-12 pull of this repo.

**What slicer are you using?**

PrusaSlicer.

**Are you using the config bundles from this repo?**

Yes.

**Did you [Adjust your slicer](https://github.com/bassamanator/Sovol-SV06-firmware#adjust-your-slicer)**?

Yes.

**Additional context**
Printing Elegoo PLA at 210C with 96% extrusion multiplier.

**Comment by bassamanator** [comment]
You need to calibrate your settings for your filament and/or adjust the macro to your needs.

**Reply by earthgrazer** [reply]
I've calibrated the retraction, extrusion, and temperature settings for my prints. The prints look good in general with just a minimal amount of angel hair in limited locations. It's just this PRINT_END macro that always leaves a thick string at the end. It's so thick that the string almost always stays attached to the nozzle after the print completes.

I've tried bumping up the `pre_purge_prime_length` var from the default 1.4 to 2, but that still hasn't eliminated the stringing.

Looking at Sovol's Cura profile, they have the following end gcode:
```
G91 ;Relative positioning
G1 E-2 F2700 ;Retract a bit
G1 E-2 Z0.2 F2400 ;Retract and raise Z
G1 X0 Y220 F3000 ;Wipe out
G1 Z10 ;Raise Z more
G90 ;Absolute positionning
G1 X0 Y{machine_depth} ;Present print
```
Looks like they're doing -2 retraction twice for a total of -4. That's quite a bit higher than the -1.4 default here. I'll give -4 a try with my next print.

**Comment by bassamanator** [comment]
After a whole year of printing I haven't encountered this issue.

Be careful to not increase the retraction too much or you will get a filament jam sooner or later.

I'd love to see a picture of this problem.
